staging/src/k8s.io/component-base/metrics/prometheus/feature/metrics.go

```diff
@@ -0,0 +1,50 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package feature
+
+import (
+	"context"
+	"fmt"
+
+	k8smetrics "k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+)
+
+var (
+	// featureInfo is a Prometheus Gauge metrics used for recording the enablement of a k8s feature.
+	featureInfo = k8smetrics.NewGaugeVec(
+		&k8smetrics.GaugeOpts{
+			Namespace:      "k8s",
```

Comment:
I would prefer `kubernetes` as a prefix to be consistent with existing general purpose metrics such as `kubernetes_build_info`.

Reply:
I already use k8s for the health check metrics. This prefix currently implies that it is linked into all Kubernetes components, but I don't believe kubernetes build info is built into each binary.

Reply:
I quickly checked on Prometheus and all the components are reporting it

Reply:
```bash
$ grep -rni "component-base/metrics/prometheus/version"                  
cmd/kube-controller-manager/controller-manager.go:30:	_ "k8s.io/component-base/metrics/prometheus/version"  // for version metric registration
cmd/kube-apiserver/apiserver.go:28:	_ "k8s.io/component-base/metrics/prometheus/version"  // for version metric registration
cmd/kubemark/hollow-node.go:44:	_ "k8s.io/component-base/metrics/prometheus/version"    // for version metric registration
cmd/kube-scheduler/scheduler.go:25:	_ "k8s.io/component-base/metrics/prometheus/version" // for version metric registration
cmd/cloud-controller-manager/main.go:38:	_ "k8s.io/component-base/metrics/prometheus/version"  // for version metric registration
cmd/kubelet/kubelet.go:30:	_ "k8s.io/component-base/metrics/prometheus/version" // for version metric registration
cmd/kube-proxy/proxy.go:24:	_ "k8s.io/component-base/metrics/prometheus/version"    // for version metric registration
vendor/modules.txt:2042:k8s.io/component-base/metrics/prometheus/version
staging/src/k8s.io/cloud-provider/sample/basic_main.go:34:	_ "k8s.io/component-base/metrics/prometheus/version"  // for version metric registration
```